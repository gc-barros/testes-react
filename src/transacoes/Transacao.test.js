import React from "react";
import { render } from '@testing-library/react';
import Transacao from "./Transacao";
import Transacoes from './Transacoes';

describe('Componente de transação do extrato', () => {
  it('O snapshot do componente deve permanecer sempre o mesmo',
  () => {
    const { container } = render (<Transacao
      data="08/09/2020"
      tipo="saque"
      valor="20.00"
    />)

    expect(container.firstChild).toMatchSnapshot();
  })

  it("Snapshot das transações deve ser renderizado", () => {
    const transacoes = [
      {
        id: 1,
        transacao: "saque",
        tipo: "saque",
        valor: "30.00",
        data: "10/12/2021",
      },
      {
        id: 2,
        transacao: "deposito",
        tipo: "deposito",
        valor: "40.00",
        data: "10/12/2021",
      },
    ];
    const { container } = render(<Transacoes transacoes={transacoes} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
