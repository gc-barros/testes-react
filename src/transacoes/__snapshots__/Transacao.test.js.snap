// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de transação do extrato O snapshot do componente deve permanecer sempre o mesmo 1`] = `
<div
  class="transacao-container"
>
  <p>
    08/09/2020
  </p>
  <p>
    saque
  </p>
  <p>
    R$ 
    20.00
  </p>
</div>
`;

exports[`Componente de transação do extrato Snapshot das transações deve ser renderizado 1`] = `
<div
  data-testid="transacoes"
>
  <div
    class="transacao-container"
  >
    <p>
      10/12/2021
    </p>
    <p>
      saque
    </p>
    <p>
      R$ 
      30.00
    </p>
  </div>
  <div
    class="transacao-container"
  >
    <p>
      10/12/2021
    </p>
    <p>
      deposito
    </p>
    <p>
      R$ 
      40.00
    </p>
  </div>
</div>
`;
